test/cpos/test_cpo_schedule.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) Lucian Radu Teodorescu
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <catch2/catch.hpp>
+#include <execution.hpp>
+
+namespace ex = std::execution;
+
+struct my_sender {
+  template <template <class...> class Tuple, template <class...> class Variant>
+  using value_types = Variant<Tuple<>>;
+  template <template <class...> class Variant>
+  using error_types = Variant<std::exception_ptr>;
+  static constexpr bool sends_done = true;
```

Comment:
Should we use a common base class for those as they are repeated quite often?

Reply:
My experience tells me that it would not have the desired effect. Most of the senders will have their own definition of these 3 members, so it's hard to abstract it out wile reducing the text size. If that could have been done, it probably  should have been in the proposal. In my experience, the extra indirection added by another abstraction compensates for the small gains you might have.

Also, after a while, you can get accustomed to this, and it's not that bad. Ignoring the `template` lines, the 3 remaining lines are quite simple.

Reply:
Fixed with Eric's proposal.